### PR TITLE
add event type (is_trigger bit) to Info

### PR DESCRIPTION
**Pull request by bmcase**
Implementing https://github.com/private-attribution/ipa/issues/593  to add an event type (is_trigger) to the associated data/Info.  Right now have it with `pub type`  and not an `enum`

### CODE REVIEW

File: src/hpke/mod.rs [hunk 1]
```diff
@@ -373,7 +424,7 @@ mod tests {
                 let mut suite = EncryptionSuite::new(10, rng.clone());
                 // keep the originals, in case if we need to damage them
                 let (mut mkp_clone, mut site_domain_clone, mut helper_clone) = (mkp_origin.clone(), site_domain.clone(), helper_origin.clone());
-                let info = Info::new(0, 0, &mkp_origin, &site_domain, &helper_origin).unwrap();
+                let info = Info::new(0, 0, EventType::Source, &mkp_origin, &site_domain, &helper_origin).unwrap();
```
Comment (akoshelev) [inline src/hpke/mod.rs:427]: this validates that source events will fail to decrypt with trigger bits, but does not validate the opposite. 

We can do both and it is easy with proptest

* generate a random bit
```rust
fn arbitrary_info_corruption(corrupted_info_field in 1..6,
                                         mkp_origin in "[a-z]{10}",
                                         site_domain in "[a-z]{10}",
                                         helper_origin in "[a-z]{10}",
                                         trigger_bit in 0_u8..=1, // <---
                                         seed: [u8; 32]) {
```

* convert this bit to event type

```rust
let event_type = EventType::try_from(trigger_bit).unwrap();
```

* flip the bit to induce trigger bit corruption

```rust
3 => Info {
    event_type: EventType::try_from(trigger_bit ^ 1).unwrap(),
    ..encryption.info
},

```

**Review by akoshelev — APPROVED** [review]
lgtm, we need to fix formatting issues and should be good to go

congrats @bmcase with your first PR!

File: src/hpke/mod.rs (hunk 1 above)
Comment (bmcase) [reply src/hpke/mod.rs]: okay, I added this to the arbitrary_info_corruption test and I think fixed the format issue

**Review by akoshelev — APPROVED** [review]

File: src/hpke/info.rs [hunk 2]
```diff
@@ -80,7 +84,8 @@ impl<'a> Info<'a> {
             + self.site_domain.len()
             + 4 // account for 4 delimiters
             + std::mem::size_of_val(&self.key_id)
-            + std::mem::size_of_val(&self.epoch);
+            + std::mem::size_of_val(&self.epoch)
+            + std::mem::size_of_val( &self.event_type);
```
[the remark this answers is not in the document]
Comment (bmcase) [reply src/hpke/info.rs:88]: I don't think the latest version has that; it should automatically update the PR, right?  https://github.com/bmcase/raw-ipa/blob/main/src/hpke/info.rs

[the remark this answers is not in the document]
Comment (bmcase) [reply src/hpke/info.rs:88]: https://github.com/private-attribution/ipa/pull/598/commits/a24fa24e182aded452874790e327231d53f3b740#diff-041e8b54bf6273bc7609d07af12d63560ddc2a5f7c9c752e6222f8cbee418d65L86-L88

[the remark this answers is not in the document]
Comment (akoshelev) [reply src/hpke/info.rs:88]: looks like it is still there :)

```rust
                 -----v-----
std::mem::size_of_val( &self.event_type);
```

[the remark this answers is not in the document]
Comment (akoshelev) [reply src/hpke/info.rs:88]: to update this PR, you would need to push your changes again: `git push`

**Pull request merged**